ai/data_objects.py

```diff
@@ -7,10 +11,10 @@ class MessageCopy:
     '''
     def __init__(
         self,
-        content=None,
-        display_name=None,
-        avatar_url=None,
-        attachments=[]
+        content: Optional[str] = None,
+        display_name: Optional[str] = None,
+        avatar_url: Optional[str] = None,
+        attachments: Iterable[Attachment] = []
```

Comment:
Nice type hinting, but it could cause merge conflicts with your other PR, which also modifies these lines but in a different manner. Which one should be considered the authoritative edit?

Reply:
I took the more formal one, that declares nullables as Optional. That is more precise and informative